### ISSUE TITLE
Is it possible to make an octoprint plugin of the wood patrens?

### ISSUE THREAD

**Issue by ischon** [MoonCactus/gcode_postprocessors#3]
I use octoprint with the CuraEngine and i would love to use your script, but i was wondering is it possible to use it this way or could you built a octoprint plugin  for this.

**Comment by MoonCactus** [comment]
Given what is stated here http://docs.octoprint.org/en/master/plugins/gettingstarted.html I suspect it would not be that difficult, may be based on this one https://github.com/ntoff/Octoprint-M84MotOff/blob/master/octoprint_m84motoff/__init__.py (it rewrites some gcode into other ones).
I may try to give it a look when I have time, but I am extremely busy for the next ... er, months?!! :/

Nb: thanks, I just did not think octoprint would support python and you made me realize it does :)

**Comment by ischon** [comment]
np I like to help people so they can help me, while you are working on this i wil use [https://www.tecrd.com/page/liens/stl_wood](url) as long as needed

**Comment by fluffy-critter** [comment]
I would also love to see this made into an Octoprint plugin. I'm tempted to try converting it myself; it seems like it shouldn't be *too* difficult.

**Comment by MoonCactus** [comment]
@fluffy-critter if you want to start writing it, I would gladly help ;)
I spent some time digging through the doc (which is well done). But octoprint did grow a lot and I can't make my mind what the best choice would be (ex. preprocessor vs on-the-fly modification of the g-code).
I also checked Cura plugins once again btw, and it still looks messy :'(

**Comment by fluffy-critter** [comment]
Yeah, the main difficulty for me is I've never written an Octoprint plugin, and I'm not sure how to best debug one without wasting a lot of filament. :)

I would think preprocessing would still be the better approach since that would allow Octoprint to still upload the complete gcode to a supported printer's SD card for standalone printing, wouldn't it?

**Comment by MoonCactus** [comment]
It is worse for me, I don't know nor use OctoPrint -- I just tried it once in 2013(?) a second time yesterday!
I could just add a user interface on top of my script but it would not help people who slice their STL from within OctoPrint.

I should register and ask in OctoPrint forum may be.
This script once was a triple "standalone" / "Skeinforge" / "Cura plugin" but Skeinforge is dead and Cura keeps breaking their own plugin format. So it ended up in wasted time :/

Now, sure I trust Gina a lot, I met her 7 years ago (ouch) and kept talking since, she's really brilliant. That's only why OctoPrint suits me better than Cura as a new candidate for an integration :)
